src/game/logic/ai/turretai.h

```diff
@@ -0,0 +1,293 @@
+
+    void Remove_Self_As_Targeter();
+    void Recenter_Turret();
+    bool Trying_To_Aim_At_Target(const Object *victim) const;
+    UpdateSleepTime Update_Turret_AI();
+    void Start_Rot_Or_Pitch_Sound();
+    void Stop_Rot_Or_Pitch_Sound();
+
+    void Set_Turret_Target_Object(Object *victim, bool force_attack);
+    void Set_Turret_Target_Position(const Coord3D *victim_pos);
+    void Set_Turret_Enabled(bool enable);
+
+    bool Is_Enabled() const { return (m_flags & 0x10) != 0; }
+    bool Is_Allows_Pitch() const { return m_data->m_isAllowsPitch; }
+
+    float Get_Turret_Angle() const { return m_angle; }
+    float Get_Turret_Pitch() const { return m_pitch; }
+    float Get_Turn_Rate() const { return m_data->m_turnRate; }
+    float Get_Natural_Turret_Angle() const { return m_data->m_naturalTurretAngle; }
+    float Get_Natural_Turret_Pitch() const { return m_data->m_naturalTurretPitch; }
+    const Object *Get_Owner() const { return m_owner; }
+    Object *Get_Owner() { return m_owner; }
+    float Get_Pitch_Rate() const { return m_data->m_pitchRate; }
+    float Get_Min_Physical_Pitch() const { return m_data->m_minPhysicalPitch; }
+    float Get_Fire_Pitch() const { return m_data->m_firePitch; }
+    float Get_Ground_Unit_Pitch() const { return m_data->m_groundUnitPitch; }
+    const Team *Get_Victim_Initial_Team() { return m_victimInitialTeam; }
+    float Get_Min_Idle_Scan_Interval() const { return m_data->m_minIdleScanInterval; }
+    float Get_Max_Idle_Scan_Interval() const { return m_data->m_maxIdleScanInterval; }
+    WhichTurretType Get_Which_Turret() const { return m_whichTurret; }
+    float Get_Min_Idle_Scan_Angle() const { return m_data->m_minIdleScanAngle; }
+    float Get_Max_Idle_Scan_Angle() const { return m_data->m_maxIdleScanAngle; }
+    float Get_Recenter_Time() const { return m_data->m_recenterTime; }
+
+    bool Is_Attack_Forced() const { return (m_flags & TURRET_ATTACK_FORCED) != 0; }
+    bool Is_Positive_Sweep() const { return (m_flags & TURRET_POSITIVE_SWEEP) != 0; }
+    bool Is_Flag80() const { return (m_flags & TURRET_80) != 0; }
+
+    void Set_Positive_Sweep(bool set) { m_flags = (TURRET_POSITIVE_SWEEP * set) | m_flags & ~TURRET_POSITIVE_SWEEP; }
+
+private:
+    enum Flags
+    {
+        TURRET_PLAY_ROTATION_SOUND = 1,
+        TURRET_PLAY_PITCH_SOUND = 2,
+        TURRET_POSITIVE_SWEEP = 4,
+        TURRET_DID_FIRE = 8,
+        TURRET_ENABLED = 0x10,
+        TURRET_FIRES_WHILE_TURNING = 0x20,
+        TURRET_ATTACK_FORCED = 0x40,
+        TURRET_80 = 0x80,
+    };
+
+    const TurretAIData *m_data;
+    WhichTurretType m_whichTurret;
+    Object *m_owner;
+    TurretStateMachine *m_turretStateMachine;
+    float m_angle;
+    float m_pitch;
+    AudioEventRTS m_turretRotOrPitchSound;
+    unsigned int m_unkFrame1;
+    Team *m_victimInitialTeam;
+    TurretTargetType m_targetType;
+    unsigned int m_unkFrame2;
+    unsigned int m_sleepUntil;
+    unsigned char m_flags;
```

Comment:
this was likely a bitfield based on xfer

Reply:
fixed.